conferencePlanner/conferencePlannerApi/Repositories/Implementations/MongoDBConferenceRepo.cs

```diff
@@ -21,50 +20,61 @@ public MongoDBConferenceRepo(IConfiguration config)
             _conferenceCollection = _database.GetCollection<Conference>("Conferences");
         }
 
-        public async Task<Conference?> CreateAsync(Conference conference)
+        public async Task<Conference> CreateAsync(Conference conference)
         {
-            conference.Id = await GetNextConferenceIdAsync();
-            _conferenceCollection.InsertOne(conference);
-            return conference;
+            Conference response = await _conferenceCollection.Find(Builders<Conference>.Filter.Eq("_id", conference.Id)).FirstOrDefaultAsync();
```

Comment:
this crashes hits the exception every time you add ID beyond the first.

fix: put check after the Id assignment, move line 28 before 25.